Chris' story: non-code contribution (#205)
kubernetes/contributor-site: content/en/blog/2021/non-code-contribution.md

---
title: "How to choose a SIG as a non-code Kubernetes contributor"
linkTitle: "How to choose a SIG as a non-code Kubernetes contributor"

> date: 2021-02-21

Author: ```suggestion
date: 2021-07-09
```

---
By [Chris Short](https://twitter.com/ChrisShort)

Kubernetes contributors aren't people in capes or part of some secret society. How to start committing to the GitHub repos that make up the project [is well documented](https://www.kubernetes.dev/docs/guide/), yet it remains intimidating for many.

A few years ago, I spoke at an event and jokingly said, "Kubernetes is just a bunch of APIs and YAML... I'm a contributor; you don't believe me?" After that talk, someone pulled me aside and asked if I was the Kubernetes contributor. They wanted to get involved in the community. Then came the real question, "I don't know which special interest group (SIG) I would work in."

> The SIG you work in depends on your skills and what you want to do with your free time.

Reviewer: or your companies time :) there should be a note here that this is the preferred route so that there is a support system in place

> The SIG you work in depends on your skills and what you want to do with your free time.

Author: ```suggestion
The SIG you work in depends on your skills and what you want to do with your (or your company's) free time.
```

My work now is focusing more on contributing to our [Contributor Marketing](https://github.com/kubernetes/community/tree/master/communication/marketing-team) subproject. The Contributor Marketing sub project's purpose is to get the word out about how to get involved. Writing this article was an output of one of those meetings. Our community does amazing things every day.

None of those things required me to write code. Non-code contributors are an essential part of our community.

# Hop into the Kubernetes community

> To join the Kubernetes community, I recommend finding a SIG where your skills align, doing things you want to do in your off time. There's a list of [Kubernetes SIGs](/resources/community-groups/); it’s had over 200 contributors over the years. If you’re not sure where to go, start with [Contributor Experience](https://github.com/kubernetes/community/tree/master/sig-contributor-experience). We’re here to help.

Reviewer: 200 contributors?

Author: Good point. I'll adjust the sentence.

Author: ```suggestion
To join the Kubernetes community, I recommend finding a SIG where your skills align, doing things you want to do in your off time. There's a list of [Kubernetes SIGs](/resources/community-groups/) with a lot of great options. If you’re not sure where to go, start with [Contributor Experience](https://github.com/kubernetes/community/tree/master/sig-contributor-experience). We’re here to help.
```

> Often, all you have to do to get started as a contributor is show up to a SIG meeting and tell folks how you want to help. Folks will be more than happy to move whatever hurdles are in your way to help you be productive. There will be some hurdles, of course--Kubernetes is a massive project--but we are all in it together.

Reviewer: being a fly on the wall first before asking what help folks need or how they can help as an individual is a better onboarding approach but this is a comment vs a review since this is about your experience

Reviewer 2: @parispittman Showing up is half the battle. I do like the fly-on-the-wall idea because that's what I've tended to do in the past. I'll make a change real quick.

My philosophy is if things were hard or confusing for me, they'd likely be hard for a large swath of folks. Process improvements can come from anyone. Why not me?

I found early on this: continually showing up to SIG meetings is one way to add value to the community. Regular participation gives you growing context in Kubernetes, and it helps in the long run.

# How I contribute

For me, contributing to Kubernetes is all about helping other contributors. I love talking to new contributors about why something is the way it is and they come up with some alternatives to doing that task. That thought exercise can be an enormously helpful contribution. That has been a part of what I've done in the community. Asking why something is the way it is can be a potent tool for improving things. I’ve helped with [new contributor workshops](https://github.com/kubernetes/community/blob/master/events/2018/05-contributor-summit/new-contributor-workshop.md), which meant showing up and sharing my knowledge. I’ve helped onboard folks into the Kubernetes GitHub organization. I’ve hosted community meetings.

> To be honest, I think the most complicated contributions to Kubernetes I’ve made involved the [community link shortener](https://github.com/kubernetes/k8s.io/blob/master/k8s.io/configmap-nginx.yaml). That consisted of editing a [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/). I had to ask for help once because something wasn’t apparent to me, I think, and there was feedback to create more consistency with already created URLs. But, at no point in time was anyone reluctant to help or the feedback negative. It was a genuinely fantastic experience. Plus, I don’t have to remember the YouTube page URL anymore. I type in [yt.k8s.io](https://yt.k8s.io). Speaking of, we have YouTube, Slack, and other admins for tools used throughout the community. They all need administration and, more importantly, participants. These things don’t necessarily require coding knowledge but familiarity with the tool that’s in use.

Reviewer 2: Good link

> To be honest, most contribution requires nothing more than Markdown and creating a pull request on GitHub. A lot of the work done in the community is in Markdown and YAML files. None of these technical skills are immediately intuitive, but it’s a lot less learning than I find most contributors know they need.

Reviewer 2: Just gonna rip it out.

My work now is focusing more on contributing to our [Contributor Marketing](https://github.com/kubernetes/community/tree/master/communication/marketing-team) subproject. The Contributor Marketing sub project's purpose is to get the word out about how to get involved. Writing this article was an output of one of those meetings. Our community does amazing things every day.

None of those things required me to write code. Non-code contributors are an essential part of our community.

# Hop into the Kubernetes community

> To join the Kubernetes community, I recommend finding a SIG where your skills align, doing things you want to do in your off time. There's a list of [Kubernetes SIGs](https://github.com/kubernetes/community/blob/master/sig-list.md); it’s had over 200 contributors over the years. If you’re not sure where to go, start with [Contributor Experience](https://github.com/kubernetes/community/tree/master/sig-contributor-experience). We’re here to help.

Reviewer 2: tyvm

Often, all you have to do to get started as a contributor is show up to a SIG meeting and tell folks how you want to help. Folks will be more than happy to move whatever hurdles are in your way to help you be productive. There will be some hurdles, of course--Kubernetes is a massive project--but we are all in it together.

> Fresh perspectives are essential to healthy projects. We need developers of all kinds to build Kubernetes and its toolchain, which is well-known. What’s lesser known is just how crucial non-code contributions are to the project’s health. If you’re interested in contributing to Kubernetes and aren’t sure how or where to get involved, feel free to join one of our [mentoring sessions](https://github.com/kubernetes/community/tree/master/mentoring). Or, if it’s easier, reach out to me directly; we’ll figure out where you belong.

Reviewer 3: We don't have any active mentoring sessions at the moment. MoC hasn't had a session since Q3 last year. I would opt to just point them towards contribex in slack.

Reviewer 3: Per contribex meeting today, MoC will be kicking back up again \o/ cc @parispittman
